### ISSUE TITLE
Unrecognized algorithm name: PBEWithSHA1AndDESede

### ISSUE THREAD

**Issue by mrrsm** [jenkinsci/docker#1112]
### Version report

Jenkins and plugins versions report:

```
Jenkins version: 2.277.4
```

- What Operating System are you using (both controller, and any agents involved in the problem)?
```
Ubuntu 18.04 - controller
```

### Reproduction steps

In the jdk 8u292 there is a bug as seen here https://bugs.java.com/bugdatabase/view_bug.do?bug_id=JDK-8266261
This got introduced in PR #1110 when the jdk got version bumped

In my particular case I am running into this while using the AzureKeyVault Plugin.  This worked/still works just fine under 8u282.
-
```
try {
  withAzureKeyvault([[ secretType: 'Certificate', name: 'credential', envVariable: 'CERT_LOCATION' ]]) {
    // Do something
  }
} catch (Exception error) {
  echo "${error}"
}
```
- Step 2...

### Results

Expected result:

The expected result is to run whatever is in the `// Do something section`

Actual result:

There is a java exception happening that the AzureKeyVault plugin is catching, wrapping, and re-throwing
```
org.jenkinsci.plugins.azurekeyvaultplugin.AzureKeyVaultException: Key protection  algorithm not found: java.security.UnrecoverableKeyException: Encrypt Private Key failed: unrecognized algorithm name: PBEWithSHA1AndDESede
```

**Comment by MarkEWaite** [comment]
Thanks @mrrsm. Is there any alternative other than switch back to JDK 8u282?

Any assessment of how widely the bug is likely to impact users?

Any indication that JDK 8u292 will receive an out of band patch or will we need to go back to 8u282 until the next quarterly release of JDK 8?

**Comment by mrrsm** [comment]
[This](https://github.com/bcgit/bc-java/issues/941#issuecomment-832008774) was the issue which pointed me at the java bug and a potential work around which I am planning on testing shortly for my specific use case.

I'm guessing that anyone doing PKCS#12 certificate work will see this issue.  Unless there is a good case to keep the old PBEWithSHA1AndDESede algorithm using a stronger algorithm for PKCS#12, assuming the workaround above works, seems like a good mitigation path while being able to stay on JDK 8u292.  If someone required PBEWithSHA1AndDESede they would be SOL as of right now unless they revert to a build with JDK 8u282.

**Comment by MarkEWaite** [comment]
Thanks @mrrsm .  I've raised a [question](https://groups.google.com/g/jenkins-infra/c/vNlgbI3vnDc/m/D_y2WJzoAgAJ) for the Jenkins infra team to get their guidance.  My current opinion is that we should continue with Jenkins using JDK 8u292 and rely on the mitigation path described in https://github.com/bcgit/bc-java/issues/941#issuecomment-832008774 and later comments.

**Comment by mrrsm** [comment]
Thanks for the prompt replies/action @MarkEWaite 

The workaround from Bouncy Castle did not work for me.  I tried a few different algorithms, so far I tried PBEWithHmacSHA256AndAES_256 and PBEWithHmacSHA256AndAES_128, and none appeared to work for my example.  My thoughts are the key/cert that is stored in KeyVault needs that specific set of instructions to decrypt/decode it.

I had to revert to version 2.277.3 to ensure our builds are working, which I have verified that they are.
I am going to keep searching for workarounds and should be able to test on and off as I find them.

**Comment by mrrsm** [comment]
I dug into my problem a bit more and figured out why the work around wouldn't work for me.

I am attempting to work on an EV Code signing cert which has the full chain in the pfx file.  The trusted root cert uses the sha1WithRSAEncryption signature algorithm which would use the removed PBEWithSHA1AndDESede class.

I believe if you try to interact with any certificates that are SHA1 signed you will hit this error, even if the certificate isn't in your chain, if it tries to validate it with certificates in your local store it will fail on SHA1 signed certs still as well as I have tested with multiple pfx files with and without the full chain in the pfx.

**Comment by mrrsm** [comment]
Per the java bug I linked in my original post.

> This issue will either be fixed by JDK-8242565 (already in OpenJDK 8u302-b04 EA build) or by JDK-8266929 (most likely going to be in 8u302-b05 EA build).

An update to jdk 8u302 once it is out of beta should solve this issue.

**Comment by MarkEWaite** [comment]
Jenkins weekly release 2.306 will include Java 8u302 in the `latest` image.  Scheduled for release Aug 10, 2021.
